### PR TITLE
Compliance card

### PR DESCRIPTION
**Pull request by mcalus3**
Add overview card with information about protected branches and license file

https://www.loom.com/share/531ada49228a495fbb2e5af3792682b4

### CODE REVIEW

**Comment by dtuite** [comment]
Mostly looks good. I will let someone else review the code.

My only ask is that missing branch protection and license is not treated as an error in the UI. It may be ok that there is no license or branch protection. The UI should inform the user of the license and branch protection situation rather than warn them.

**Review by adamdmharvey — COMMENTED** [review]
Looks like an exciting new feature!  If I'm not mistaken, this will only support GitHub.com and not GHE instances, is that right? (don't see loading of baseUrl nor passing it to Octokit, like the other widgets?) Just thought I'd ask!

**Comment by mcalus3** [comment]
> 
> 
> Mostly looks good. I will let someone else review the code.
> 
> My only ask is that missing branch protection and license is not treated as an error in the UI. It may be ok that there is no license or branch protection. The UI should inform the user of the license and branch protection situation rather than warn them.

I assumed that if someone pulls the compliance card into his backstage instance, then compliance is an important thing for him. And then lack of thereof would be somewhat serious.

It could be yellow severity alert:
image

**Comment by mcalus3** [comment]
@dtuite New visual for the missing branch/license:
image
@adamdmharvey I think the code now should support GHE repositories

File: src/components/Widgets/ComplianceCard/ComplianceCard.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  useRepoLicence,
+} from '../../../hooks/useComplianceHooks';
+import { useProjectEntity } from '../../../hooks/useProjectEntity';
+import WarningIcon from '@material-ui/icons/ErrorOutline';
+
+type ReleaseCardProps = {
+  entity: Entity;
+};
+
+const ComplianceCard: FC<ReleaseCardProps> = ({ entity }) => {
```
Comment (adamdmharvey) [inline src/components/Widgets/ComplianceCard/ComplianceCard.tsx:32]: PS: I would believe avoiding FC my apply here, though I understand this is Roadie code and not straight Backstage's project code, but aligning with Backstage ADRs may be preferred?

https://backstage.io/docs/architecture-decisions/adrs-adr006

(and sorry @mcalus3, I'm definitely not a TypeScript expert, learning as I go!! hope you are OK with the feedback 😄 )

Comment (mcalus3) [reply src/components/Widgets/ComplianceCard/ComplianceCard.tsx:32]: You are right, will change it in the next commit :)

**Comment by mcalus3** [comment]
> 
> 
> @mcalus3 let's reduce the verbosity please. I would like it to be scannable. Something like this is fine.
> 	
> Protected Branches 	⛔️ None
> License 	⛔️ None

image

**Review by iain-b — APPROVED** [review]

**Comment by iain-b** [comment]
As a further improvement we could link to the branch protection configuration page for the repo where there are none set?

**Comment by mcalus3** [comment]
> 
> 
> As a further improvement we could link to the branch protection configuration page for the repo where there are none set?

I had a link to the repo there, but @dtuite wanted it to be minimal to not hijack the overview space from more important things.

**Pull request merged**